treeopt/treeOpt.py

```diff
@@ -0,0 +1,277 @@
+        """
+        Sets the Method that is used to approximate the system responce in the
+        design space
+        :param method: one of the functions in modules/metamodell.py
+        :type method: python function
+        :return: Nothing
+        :rtype: None
+
+        """
+
+        self.smMethod = method
+
+    def set_accuracy_criterion(self, method):
+        """
+        Sets the method that is used to define the termination condition of the
+        optimization workflow
+        :param method: one of the functions in modules/accuracy
+        :type method: Python Function
+        :return: Nothing
+        :rtype: None
+
+        """
+
+        self.accuracyCriterion = method
+
+    # Starts one simulation run
+    def simulate(self, x):
+        """
+        Function that starts a Simulation and returns the resonce of the system
+        :param x: Numpy array representing a set of parameters to be written on
+        the simulation Input file
+        :type x: Numpy array
+        :return: Numpy array containing the system responce
+        :rtype: Numpy array
+
+        """
+        try:
+            if self.simulateMethod == simulate.simulate_benchmark_function:
+                return self.simulateMethod(self.benchmarkingProblem, x)
+            if self.simulateMethod == simulate.simulateExternalProgramm:
+                return self.simulateMethod(
+                    self.inputFile,
+                    self.outputFile,
+                    self.simFile,
+                    self.program,
+                    x,
+                )
+        except TypeError:
+            logging.error(traceback.format_exc())
+            print("Check your parameters")
+
+    def start_optimization(self):
+        """
+        Function that starts the previosly parameterized adaptive optimization
+        loop
+        :return: Nothing
+        :rtype: None
+
+        """
+
+        self.x = self.samplingMethod(self.limits, self.numDOE)
+        self.y = self.simulate(np.atleast_2d(self.x[0]))
+
+        for xi in self.x[1:]:
+            self.append_y_data(self.simulate(np.atleast_2d(xi)))
+
+        self.write_data(self.x, "DoeData")
+        self.write_data(self.y, "DoeResponce")
+
+        self.optGoal = False
+
+        self.ite = 0
+        self.maxite = 5
+
+        while self.optGoal is False:
+
+            self.sm = self.smMethod(self.x, self.y)
+
+            self.nX = optimize.get_lowest_variance(self.sm, self.limits)
+
+            self.append_x_data(self.nX)
+            self.append_y_data(self.simulate(np.atleast_2d(self.nX)))
+
+            self.write_data(self.x, "DoeData")
+            self.write_data(self.y, "DoeResponce")
+
+            self.ite += 1
+
+            if self.ite == self.maxite:
+                self.optGoal = True
+
+            self.current_best_point = optimize.find_minimum(
+                self.sm, self.limits
+            )
+            print("bester Punkt:", self.current_best_point)
```

Comment:
avoid `print` in a library, you don't know if it is used from something that has a console. We could use a logging library for this. Also stick to English.